w3af/core/data/context/context/javascript.py

```diff
@@ -36,7 +40,25 @@ class ScriptMultiLineComment(BaseContext):
 
 
 class ScriptStringGeneric(BaseContext):
-    pass
+    ATTR_DELIMITER = None
+
+    def can_break(self):
+        """
+        :return: True if we can break out
+        """
+        escaped = False
+        for c in self.payload:
+            if c == '\\':
+                escaped = not escaped
+            elif c == self.ATTR_DELIMITER and not escaped:
+                return True
+            else:
+                escaped = False
+
+        return False
+
+    def get_payloads(self):
+        return {self.ATTR_DELIMITER, '\\%s' % self.ATTR_DELIMITER}
```

Comment:
Ah! Tested it and it works as you said. I expected browsers to use `\"` as escape of the `"`

My test was with chrome and `<a x="xxx\" href=\"/foobar\" yyy">test</a>`

I still don't understand why you need to handle those escapes. The old version of the code should also support them. By support I mean that it was possible to find XSS? We should chat over GTalk so you can explain these changes to me.